go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -407,8 +407,9 @@ func getEventInfoForSpark(sj *sparkOp.SparkApplication) (*pluginsCore.TaskInfo,
 			})
 		}
 	} else if sj.Status.AppState.State == sparkOp.RunningState && sj.Status.DriverInfo.WebUIIngressAddress != "" {
-		// Append https as the operator doesn't currently.
-		customInfoMap[sparkDriverUI] = fmt.Sprintf("https://%s", sj.Status.DriverInfo.WebUIIngressAddress)
+		// Older versions of spark-operator does not append http:// but newer versions do.
+		uri := strings.TrimPrefix(sj.Status.DriverInfo.WebUIIngressAddress, "http://")
+		customInfoMap[sparkDriverUI] = fmt.Sprintf("https://%s", uri)
```

Comment:
To maintain backwards compatibility does it make sense to check for the prefix rather than removing it (noop if it doesn't exist) and then appending it? Maybe something like

```
uri := sj.Status.DriverInfo.WebUIIngressAddress
if !strings.HasPrefix("https://") {
    uri = fmt.Sprintf("https://%s", uri)
}
customInfoMap[sparkDriverUI] = uri
```